### PR TITLE
CIV-6131 Claimant response task list-P1._Snyk

### PR DESCRIPTION
**Pull request by laperla11**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-6131

### Change description ###
- Developed the framework for Claimant Intent Task List
- Display the View the defendant's response, Hearing requirements and Submit response tasks
- Fixed the redirection bug from DQ - permission expert page
- Fixed the redirection bug on DQ -welsh language page

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/routes/features/claimantResponse/reviewDefendantsResponseController.ts [hunk 1]
```diff
@@ -38,19 +38,17 @@ reviewDefendantsResponseController.post(CLAIMANT_RESPONSE_REVIEW_DEFENDANTS_RESP
...
     if (claim?.responseStatus === ClaimResponseStatus.PA_NOT_PAID_PAY_BY_DATE) {
       const financialDetails = getFinancialDetails(claim, lang);
       res.render('features/claimantResponse/how-they-want-to-pay-response', {
         claim,
-        continueLink,
         financialDetails,
         paymentDate: formatDateToFullDate(claim.partialAdmission.paymentIntention.paymentDate, lang),
       });
     } else {
+      await saveClaimantResponse(claimId, true, 'defendantResponseViewed');
```
Comment (laarmada) [inline src/main/routes/features/claimantResponse/reviewDefendantsResponseController.ts:49]: use a constant instead of text: 'defendantResponseViewed'

**Review by laarmada — APPROVED** [review]

File: src/main/services/features/response/taskListService.ts [hunk 2]
```diff
@@ -12,9 +12,6 @@ import {getLng} from '../../../common/utils/languageToggleUtils';
 import {t} from 'i18next';
 import {Claim} from '../../../common/models/claim';
 
-let completed = 0;
-let total = 0;
-
```
Comment (laarmada) [inline src/main/services/features/response/taskListService.ts:17]: move the file one level above?

**Review by GorilaNaranja — APPROVED** [review]

File: src/main/routes/features/claimantResponse/claimantResponseTasklistController.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import {Router} from 'express';
+import {CLAIMANT_RESPONSE_TASK_LIST_URL} from '../../urls';
+import {getDescription, getTitle} from '../../../services/features/response/taskListService';
```
[the remark this answers is not in the document]
Comment (laperla11) [reply src/main/routes/features/claimantResponse/claimantResponseTasklistController.ts]: resolved

File: src/main/services/features/response/taskListService.ts (hunk 2 above)
Comment (laperla11) [reply src/main/services/features/response/taskListService.ts:17]: resolved

File: src/main/routes/features/claimantResponse/reviewDefendantsResponseController.ts (hunk 1 above)
Comment (laperla11) [reply src/main/routes/features/claimantResponse/reviewDefendantsResponseController.ts]: resolved

File: src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts [hunk 4]
```diff
@@ -29,4 +30,58 @@ export class DirectionQuestionnaire {
...
+      this.nothingExpertCanExamine ||
+      this.isExpertDetailsAvailable) {
+      return true;
+    }
+    return false;
+  }
+
+  get isUnavailabilityDatesCompleted(): boolean {
+    // TODO : include completion logic for unavailable dates when `unavailable-for-hearing` page is developed
+    if (!this.hearing?.whyUnavailableForHearing) {
```
Comment (LeonardoPalmeiro) [inline src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts:82]: Might be an option to change this with ternary operator.
What do you thing about it?

File: src/main/routes/features/directionsQuestionnaire/experts/expertSmallClaimsController.ts [hunk 5]
```diff
@@ -1,17 +1,31 @@
...
-  res.render(expertSmallClaimsViewPath,
-    {
-      giveEvidenceYourselfScreenURL: constructResponseUrlWithIdParams(claimId, DQ_GIVE_EVIDENCE_YOURSELF_URL),
-      reportWrittenByExpertScreenURL: constructResponseUrlWithIdParams(claimId, DQ_EXPERT_REPORT_DETAILS_URL),
-    });
+  res.render(expertSmallClaimsViewPath);
+});
+
+expertSmallClaimsController.post(DQ_EXPERT_SMALL_CLAIMS_URL, async (req: Request, res: Response, next: NextFunction) => {
+  try {
```
Comment (LeonardoPalmeiro) [inline src/main/routes/features/directionsQuestionnaire/experts/expertSmallClaimsController.ts:20]: remove extra empty space.

**Review by GorilaNaranja — DISMISSED** [review]

File: src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts [hunk 6]
```diff
@@ -29,4 +30,58 @@ export class DirectionQuestionnaire {
...
+      this.hearing?.determinationWithoutHearing &&
+      this.isExpertJourneyCompleted &&
+      this.defendantYourselfEvidence &&
+      this.witnesses?.otherWitnesses &&
+      this.isUnavailabilityDatesCompleted &&
+      this.hearing?.phoneOrVideoHearing &&
+      this.vulnerabilityQuestions?.vulnerability &&
+      this.hearing?.supportRequiredList &&
+      this.hearing?.specificCourtLocation &&
+      this.welshLanguageRequirements?.language
```
Comment (LeonardoPalmeiro) [inline src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts:60]: please, could you add some unit test in this? if it is possible.

**Review by laarmada — APPROVED** [review]

File: src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts (hunk 4 above)
Comment (LeonardoPalmeiro) [reply src/main/common/models/directionsQuestionnaire/directionQuestionnaire.ts:82]: as well added unit test in here

**Review by LeonardoPalmeiro — CHANGES_REQUESTED** [review]

**Review by GorilaNaranja — DISMISSED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]
LGTM

**Review by GorilaNaranja — APPROVED** [review]

**Review by laarmada — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Pull request merged**